content/systems/filecoin_vm/runtime/gascost/_index.md

```diff
@@ -9,4 +9,79 @@ dashboardTests: 0
 # VM Gas Cost Constants
 ---
 
-{{<embed src="vm_gascosts.go" lang="go" >}}
+Every operation that triggers computation or storage on the Filecoin VM incurs payment in terms of `GasCharge` (`VirtualCompute`, `VirtualStorage`). Charges can also be incurred by actors (`ComputeGas`, `StorageGas`). The `GasCharge struct` includes the `Name` of the cost to be incurred, which is taken by the `Pricelist`.
+
+```go
+type GasCharge struct {
+	Name  string
+	Extra interface{}
+
+	ComputeGas int64
+	StorageGas int64
+
+	VirtualCompute int64
+	VirtualStorage int64
```

Comment:
This virtual stuff is Lotus-specific implementation details related to testing and the setting of gas prices.